pkg/slayers/path/scion/decoded.go

```diff
@@ -207,6 +207,7 @@ func (s *Decoded) SerializeTo(b []byte /*@, ghost ubuf []byte @*/) (r error) {
 // @ ensures  r == nil ==> p.Mem(ubuf)
 // @ ensures  r == nil ==> p == s
 // @ ensures  r == nil ==> typeOf(p) == type[*Decoded]
+// @ ensures  r == nil && old(unfolding s.Mem(ubuf) in s.Base.InfValid()) ==> unfolding (p.(*Decoded)).Mem(ubuf) in (p.(*Decoded)).Base.InfValid()
```

Comment:
why do we use unfolding here when we could use the private functions you defined for Decoded?